pkg/vdr/key/creator.go

```diff
@@ -111,13 +126,21 @@ func getKeyCode(keyType kms.KeyType, verificationMethod *did.VerificationMethod)
 }
 
 func fetchECKeyCodeFromVerMethod(method *did.VerificationMethod) (uint64, error) {
-	ecdsaCodesByKeyLen := map[int]uint64{
-		64:  fingerprint.P256PubKeyMultiCodec,
-		96:  fingerprint.P384PubKeyMultiCodec,
-		132: fingerprint.P521PubKeyMultiCodec,
+	// ec keys must pass in a jwk
+	if method.JSONWebKey() == nil {
+		return 0, fmt.Errorf("jsonWebKey is required for EC key")
 	}
 
-	return ecdsaCodesByKeyLen[len(method.Value)], nil
+	switch method.JSONWebKey().Crv {
```

Comment:
The point of the map is to avoid switch/case.. if you want to use the curve type instead of key size, then replace the map keys to be curves instead of key size. This way you can avoid doing switch/case and use the map directly.

Reply:
I'm not sure what the preference is of avoiding a switch statement, but if I revert to using a map then an additional if statement will be required in case the curve is not recognized at all. Note that this was an issue with the prior implementation as well in that when the key length did not match any of the values the function was just returning 0 and no error.

If you still prefer a map and an if statement then I can change it to that.

Reply:
yes, with a map, we'd still want to check if the value is found, otherwise return an error, it was probably missed before.

you can keep switch/case then for now, no need for additional changes.